File: src/lc/p1/java/TwoSumTest.java
import org.junit.runner.JunitCore;
import org.junit.runner.Result;
import org.junit.runner.notification.Failure;


public class TwoSumTest{
    @BeforeClass
    public void twoSumTestSetupClass(){
        this.solution = TwoSum();
    }

    @Test
    public void twoSumTestCase1(){
        int[] nums = {2, 7, 11, 15};
        int target = 9; 
        int[] ret = {0, 1};
        assertEqual(ret, this.solution.twosum1(nums, target));
        assertEqual(ret, this.solution.twosum2(nums, target));
    }

    public static void main(String[] args){
        Result result = JunitCore.runClasses(TwoSumTest.class);
        for(Failure failure : result.getFailures()){
            System.out.println(failure.toString());
        }
    }
}

File: src/lc/p1/cpp/twoSumTest.cpp
#include"twoSumSolution.cpp"
#include<iostream>
using namespace std;

int main(){
    vector<int> nums={2, 7, 11, 15};
    int target(9);
    vector<int> res={0, 1};
    Solution solution;
    vector<int> ret = solution.twoSum(nums, target);

    if(ret == res){
        cout << "Passed test" << endl;
    }else{
        cout << "Failed test" << endl;
    }
}